dali/server/daldap.cpp

```diff
@@ -125,18 +127,23 @@ class CDaliLdapConnection: implements IDaliLdapConnection, public CInterface
         else
         {
             DBGLOG("NULL UserDescriptor in daldap.cpp getPermissions('%s')", key);
-        }
 
-        if (0 == username.length())
-        {
+            // If no user was provided, try to use the default user
+            if (disableFilesDefaultUser || filesdefaultuser.isEmpty())
+            {
+                return SecAccess_None; // no access if no default user or disabled
+            }
+
             username.append(filesdefaultuser);
             decrypt(password, filesdefaultpassword);
-            OWARNLOG("Missing credentials, injecting deprecated filesdefaultuser for request %s %s", key, nullText(obj));
+            OWARNLOG("Missing credentials, injecting deprecated filesdefaultuser (%s) for request %s %s", filesdefaultuser.str(), key,
+                     nullText(obj));
             logNullUser(nullptr);
         }
 
         Owned<ISecUser> user = ldapsecurity->createUser(username);
-        user->setAuthenticateStatus(AS_AUTHENTICATED);
+        user->credentials().setPassword(password);
+        //  user->setAuthenticateStatus(AS_AUTHENTICATED);  let normal user authentication take place
```

Comment:
I don't think this should be changed.
It was deliberate to treat calls to Dali getPermissions as from a trusted user.
See : https://hpccsystems.atlassian.net/browse/HPCC-22415

Reply:
I made an update that sets a passed in user as authenticated, but forces authentication for the default user. Otherwise any user/pw combination would have access in these cases.